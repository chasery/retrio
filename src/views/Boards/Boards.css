.Boards__wrapper {
  padding: 1rem 1.5rem;
}

.Boards__header {
  display: flex;
}
.Boards__header > h2 {
  flex-grow: 1;
}

@media all and (min-width: 50rem) {
  .Boards__wrapper {
    max-width: 64rem;
    align-self: center;
    margin: 0 auto;
  }
}
