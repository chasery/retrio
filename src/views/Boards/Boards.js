import React from 'react';
import { Link } from 'react-router-dom';
import Header from '../../components/Header/Header';
import BoardsList from '../../components/BoardsList/BoardsList';
import './Boards.css';

function Boards(props) {
  // Response object I am expecting for boards
  const boards = [
    {
      id: 1,
      name: 'Deploy Happy Retrio Board',
      created_at: new Date(),
      modified_at: new Date(),
    },
    {
      id: 2,
      name: 'Someone Else Broke It Retrio Board',
      created_at: new Date(),
      modified_at: null,
    },
    {
      id: 3,
      name: 'Team Name Pending Retrio Board',
      created_at: new Date(),
      modified_at: new Date(),
    },
  ];

  return (
    <>
      <Header />
      <main role='main'>
        <section className='Boards'>
          <div className='Boards__wrapper'>
            <div className='Boards__header'>
              <h2>My Boards</h2>
              <Link to='/boards/add-board'>Add Board</Link>
            </div>
            <BoardsList boards={boards} />
          </div>
        </section>
      </main>
    </>
  );
}

export default Boards;
