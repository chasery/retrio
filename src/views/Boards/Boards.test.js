import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Boards from './Boards';

describe('Boards component', () => {
  it('renders without error', () => {
    const wrapper = shallow(<Boards />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
