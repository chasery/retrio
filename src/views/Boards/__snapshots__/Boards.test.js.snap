// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Boards component renders without error 1`] = `
<Fragment>
  <Header />
  <main
    role="main"
  >
    <section
      className="Boards"
    >
      <div
        className="Boards__wrapper"
      >
        <div
          className="Boards__header"
        >
          <h2>
            My Boards
          </h2>
          <Link
            to="/boards/add-board"
          >
            Add Board
          </Link>
        </div>
        <BoardsList
          boards={
            Array [
              Object {
                "created_at": 2021-04-22T18:26:51.112Z,
                "id": 1,
                "modified_at": 2021-04-22T18:26:51.112Z,
                "name": "Deploy Happy Retrio Board",
              },
              Object {
                "created_at": 2021-04-22T18:26:51.112Z,
                "id": 2,
                "modified_at": null,
                "name": "Someone Else Broke It Retrio Board",
              },
              Object {
                "created_at": 2021-04-22T18:26:51.112Z,
                "id": 3,
                "modified_at": 2021-04-22T18:26:51.112Z,
                "name": "Team Name Pending Retrio Board",
              },
            ]
          }
        />
      </div>
    </section>
  </main>
</Fragment>
`;
